pymemcache/client/retrying.py

```diff
@@ -0,0 +1,178 @@
+""" Module containing the RetryingClient wrapper class. """
+
+from time import sleep
+
+
+def _ensure_tuple_argument(argument_name, argument):
+    """
+    Helper function to ensure the given arguments are tuples of Exceptions (or
+    subclasses), or can at least be converted to such.
+
+    Args:
+      argument_name: str, name of the argument we're checking, only used for
+        raising meaningful exceptions.
+      argument: any, the argument itself.
+
+    Returns:
+      tuple[Exception]: A tuple with the elements from the argument if they are
+        valid.
+
+    Exceptions:
+      ValueError: If the argument was not None, tuple or Iterable.
+      ValueError: If any of the elements of the argument is not a subclass of
+        Exception.
+    """
+
+    # Ensure the argument is a tuple, set or list.
+    if argument is None:
+        return tuple()
+    elif not isinstance(argument, (tuple, set, list)):
+        raise ValueError("%s was not a tuple or Iterable." % argument_name)
+
+    # Convert the argument before checking contents.
+    argument_tuple = tuple(argument)
+
+    # Check that all the elements are actually inherited from Exception.
+    # (Catchable)
+    if not all([issubclass(arg, Exception) for arg in argument_tuple]):
+        raise ValueError(
+            "%s contained elements other than Exceptions." % argument_name
```

Comment:
Some kind of remark here.

The error message could be more detailed and it could bring more details about why the users face it and what are the solutions for them. We should be proactive and lead users on the right path to avoid mistakes and so reduce the human support in a second time (github issues, etc...).

Reply:
I'm still not sold on super wordy exception messages, so I'll let a collaborator weigh in on that one.
I've reworded the exception to be more specific about what is required though.
https://github.com/martinnj/pymemcache/commit/6bd21d0261f57d54c8c3d1211088bc27f07f5ee3